### PR TITLE
265 adding ipt required fields to metadata form

### PR DESCRIPTION
**Pull request by sjbruce**
Added a biological tab for Geographic Description, Sampling Description, Study Extent and Method Steps (repeating section).

Updated EML generation to include new elements.

### CODE REVIEW

**Comment by n-a-t-e** [comment]
This looks good! 

It would be good to give this "Add Item" button some context, like if items are method steps, maybe this can be in a section called "Methods" and items can be "Steps" or something like that.

Screen Shot 2023-11-30 at 4 48 05 PM

**Comment by sjbruce** [comment]
Thanks, only getting back to this now.

As suggested, I've renamed the buttons to give more context and moved the method steps themselves into a section with descriptive text.  I've also added key to the steps so they can be enumerated on the page.

I guess this could be reworked to behave like contacts or instruments - for the moment I've modeled it after the resources tab.

**Comment by n-a-t-e** [comment]
Looks good! Tiny thing - looks like "Move up" is correctly greyed out when theres a single method step, but "Move down" is still clickable (doesn't do anything)

Screen Shot 2023-12-14 at 3 07 51 PM

File: src/utils/emlTemplate.j2 [hunk 1]
```diff
@@ -89,40 +92,65 @@
...
-      </description>
-      <maintenanceUpdateFrequency>asNeeded</maintenanceUpdateFrequency>
-    </maintenance>
 
+        <methods>
+            {% for method in methodList %}
+            <methodStep>
+                <description>
+                    <section>
+                        <title>{{ method.title.en }}</title>
```
Comment (n-a-t-e) [inline src/utils/emlTemplate.j2:101]: My text editor (VSCode) is complaining about <title> being here. do you have any links to info on this <title> tag?

Comment (n-a-t-e) [reply src/utils/emlTemplate.j2:101]: also possible my editor is using the wrong schema files, if this works in the IPT thats good enough

Comment (sjbruce) [reply src/utils/emlTemplate.j2:101]: I consulted a few different sources but ultimately based my structure on the following example: https://ediorg.github.io/data-package-best-practices/methods.html

Based on the idea that the description for a methodStep element is a section and a section has a title.

This seems to bear out in the EML schema detailed here, but I'm no expert in this either - the following validator doesn't complain about it: https://knb.ecoinformatics.org/emlparser/

It does have some opinions on roles being specified for associated party: 
`The content of element 'associatedParty' is not complete. One of '{userId, role}' is expected.`

File: src/components/FormComponents/MethodSteps.jsx
Comment (fostermh) [inline src/components/FormComponents/MethodSteps.jsx:1]: At first glance, there appears to be quite a bit of overlap between 'Method' from EML and 'Lineage' from iso1911.  Specifically the processing step.
- https://wiki.esipfed.org/LI_Lineage
- https://wiki.esipfed.org/LI_ProcessStep

File: src/utils/emlTemplate.j2 [hunk 2]
```diff
@@ -89,40 +92,65 @@
...
+                        <para>{{ method.description.en }}</para>
+                    </section>
+                </description>
+            </methodStep>
+            {% endfor %}
+            <sampling>
```
Comment (fostermh) [inline src/utils/emlTemplate.j2:107]: The sampling section seems to overlap with the following iso19115 concepts:
- https://wiki.esipfed.org/EX_Extent
- https://wiki.esipfed.org/EX_TemporalExtent
- https://wiki.esipfed.org/EX_GeographicDescription
- https://wiki.esipfed.org/MD_MaintenanceInformation

File: src/utils/emlTemplate.j2 [hunk 3]
```diff
@@ -89,40 +92,65 @@
...
+                <studyExtent>
+                    <description>
+                        <para>{{ record.biological.studyExtent.en }}</para>
+                    </description>
+                </studyExtent>
+                <samplingDescription>
```
Comment (fostermh) [inline src/utils/emlTemplate.j2:113]: samplingDescription also seems to overlap with the iso lineage section

File: src/utils/emlTemplate.j2 [hunk 4]
```diff
@@ -70,6 +72,7 @@
         </distribution>
         <coverage>
             <geographicCoverage>
+                <geographicDescription>{{ record.biological.geographicDescription.en }}</geographicDescription>
```
Comment (fostermh) [inline src/utils/emlTemplate.j2:75]: overlap with iso https://wiki.esipfed.org/EX_GeographicDescription

**Review by fostermh — COMMENTED** [review]
highlighting a few areas I think biological data and iso19115 schema overlap

**Comment by fostermh** [comment]
replaced by https://github.com/cioos-siooc/metadata-entry-form/pull/322

**Pull request closed without merging**